docs/concepts/native-interop.rst

```diff
@@ -0,0 +1,260 @@
+Native interoperability
+=======================
+
+In this document we will dive a little bit deeper into all three ways of doing 
+"native interoperability" that are available on the .NET platform. 
+
+To refresh your memory, the two main ways of doing native interoperability are:
+
+* Platform Invoke (P/Invoke)
+* COM interoperability
+
+You might be wondering why would you ever want or need to do something like this. 
```

Comment:
Is this paragraph necessary? It's accurate. I just wonder if we could go right into the reasons, which is really "few" and not "couple".